schemas/iso19139/src/main/plugin/iso19139/layout/layout-custom-fields.xsl

```diff
@@ -137,4 +137,106 @@
       </xsl:with-param>
     </xsl:call-template>
   </xsl:template>
+
+
+  <!-- SDS: Category. Render an anchor as select box populating it with a codelist -->
+  <xsl:template mode="mode-iso19139" priority="2000" match="gmd:dataQualityInfo/*/gmd:report/gmd:DQ_DomainConsistency/gmd:result/gmd:DQ_ConformanceResult/gmd:specification/gmd:CI_Citation/gmd:title/gmx:Anchor[/root/gui/currTab/text()='inspire_sds']">
+    <xsl:call-template name="render-element">
+      <xsl:with-param name="label" select="/root/gui/strings/category"/>
+      <xsl:with-param name="value" select="@xlink:href"/>
+      <xsl:with-param name="cls" select="local-name()"/>
+      <xsl:with-param name="xpath" select="gn-fn-metadata:getXPath(.)"/>
+      <xsl:with-param name="type" select="'select'"/>
+      <xsl:with-param name="listOfValues" select="/root/gui/schemas/iso19139/codelists/codelist[@name='INSPIRE_category']"/>
+      <xsl:with-param name="name" select="concat(gn:element/@ref,'_xlinkCOLONhref')"/>
+      <xsl:with-param name="editInfo" select="*/gn:element"/>
+      <xsl:with-param name="parentEditInfo" select="gn:element"/>
+      <xsl:with-param name="isDisabled" select="false()"/>
+    </xsl:call-template>
+
+    <script>
+        $( document ).ready(function(){
```

Comment:
I think we should avoid to have any JS in XSLTs. I would move that to an angular directive. One option is to pass the directive to render-element (eg. https://github.com/geonetwork/core-geonetwork/blob/develop/schemas/iso19139/src/main/plugin/iso19139/layout/layout-custom-fields.xsl#L136) or for more customization build the form here (eg https://github.com/geonetwork/core-geonetwork/blob/develop/schemas/iso19139/src/main/plugin/iso19139/layout/layout-custom-fields.xsl#L64).